### ISSUE TITLE
messages set to priority 'high' are not sent by command when the original sender is destroyed.

### ISSUE THREAD

**Issue by MoerasGrizzly** [scp-fs2open/fs2open.github.com#5968]
The send-message sexp mentions that:
> High priority are not interrupted by anything, and will be sent by Command if the sender is destroyed.

As of FSO 23.2.1 and 24.0, this is no longer the case. I do remember that this worked as described in earlier versions.

**Comment by MoerasGrizzly** [comment]
I do also vaguely recall that the <any wingman> setting would flat out not send a message if a wingman with the matching persona could not be found, even when the message priority is set to 'high'. I'd prefer it if that particular option was a little bit more robust too.

**Comment by Goober5000** [comment]
The messaging system is definitely supposed to fall back to Command if a High priority message needs to be sent with no wingman available.  This is used in FS1 and in the Port.  I don't recall offhand if the behavior is supposed to be different for `<any wingman>` vs a specific wingman.

**Comment by Goober5000** [comment]
Well I'm getting inconsistent results testing send-message on retail FS1 and retail FS2.  More investigation is needed.

**Comment by Goober5000** [comment]
> As of FSO 23.2.1 and 24.0, this is no longer the case. I do remember that this worked as described in earlier versions.

@MoerasGrizzly When was the last version in which this happened as described?

**Comment by Goober5000** [comment]
Ok it turns out that `send-message` still works correctly.  The idiosyncratic detail here is that the High messages are indeed sent by Command, but *the personas are still checked*.  If the High priority message has a persona, this persona is converted to Command if and only if a) it's an in-mission message, and b) the file has a `1_`, `2_`, etc. prefix that can be converted to an existing file with a prefix `TC_`.  On the other hand, if the High priority message has *no* persona assigned, the message will be sent regardless.

Incidentally, there is no difference in handling between `<any wingman>` and a specific wingman.

See also #4150, #4078, #4357, and their linked issues.

**Comment by MoerasGrizzly** [comment]
> > As of FSO 23.2.1 and 24.0, this is no longer the case. I do remember that this worked as described in earlier versions.
> 
> @MoerasGrizzly When was the last version in which this happened as described?

22.0 or 22.2

**Comment by Goober5000** [comment]
Well here is the mission I used for testing.  You will need to also copy TC_arrival.wav into the `voice/special` folder.
[0send-message-test.fs2.txt](https://github.com/scp-fs2open/fs2open.github.com/files/14152589/0send-message-test.fs2.txt)

Press 1, 2, or 3 to trigger messages from `<any wingman>` using the Wingman 1, 2, or 3 personas (which correspond to Beta 1, 2, and 3).  Press A to send a message from Beta 1 specifically with the Wingman 1 persona.  Press E to send a message from the Elysium with no persona.

You can also restart the mission and press Z to self-destruct every ship, then press each key again.  Pressing E will send a message from Command rather than the dead Elysium.  If you have copied TC_arrival.wav, then pressing A or 1 will send a message from Command rather than the dead Beta 1.

In summary, the number-to-TC feature will do what you need, but it is designed to work only in specific conditions.  You might also want to look at the SendMessageAltSEXPs here:  
https://github.com/FSO-Scripters/fso-scripts/tree/master/SendMessageAltSEXPs

**Comment by Goober5000** [comment]
Note that if you load the first non-training mission of FSPort, then ~K Alpha 2, Alpha 2's messages will be send by Command.

**Comment by MoerasGrizzly** [comment]
> Well here is the mission I used for testing. You will need to also copy TC_arrival.wav into the `voice/special` folder. [0send-message-test.fs2.txt](https://github.com/scp-fs2open/fs2open.github.com/files/14152589/0send-message-test.fs2.txt)
> 
> Press 1, 2, or 3 to trigger messages from `<any wingman>` using the Wingman 1, 2, or 3 personas (which correspond to Beta 1, 2, and 3). Press A to send a message from Beta 1 specifically with the Wingman 1 persona. Press E to send a message from the Elysium with no persona.
> 
> You can also restart the mission and press Z to self-destruct every ship, then press each key again. Pressing E will send a message from Command rather than the dead Elysium. If you have copied TC_arrival.wav, then pressing A or 1 will send a message from Command rather than the dead Beta 1.
> 
> In summary, the number-to-TC feature will do what you need, but it is designed to work only in specific conditions. You might also want to look at the SendMessageAltSEXPs here: https://github.com/FSO-Scripters/fso-scripts/tree/master/SendMessageAltSEXPs

I can work with that, and the easy fix is to just unassign personas from those messages that are pretty vital.
